keras/utils/csv_dataset_utils.py

```diff
@@ -0,0 +1,367 @@
+import numpy as np
+
+from keras.api_export import keras_export
+from keras.utils import dataset_utils
+from keras.utils.module_utils import tensorflow as tf
+from keras.utils.module_utils import tensorflow_io as tfio
+
+ALLOWED_FORMATS = (".csv",)
+
+
+@keras_export("keras.utils.csv_dataset_from_directory")
+def csv_dataset_from_directory(
+    directory,
+    labels="inferred",
+    label_mode="int",
+    class_names=None,
+    batch_size=32,
+    ragged=False,
+    shuffle=True,
+    seed=None,
+    validation_split=None,
+    subset=None,
+    follow_links=False,
+    stride=1,
+    head=True,
+):
+    """Generates a `tf.data.Dataset` from csv files in a directory.
+
+    If your directory structure is:
+
+    ```
+    main_directory/
+    ...class_a/
+    ......a_data_1.csv
+    ......a_data_2.csv
+    ...class_b/
+    ......b_data_1.csv
+    ......b_data_2.csv
+    ```
+
+    Then calling `csv_dataset_from_directory(main_directory,
+    labels='inferred')`
+    will return a `tf.data.Dataset` that yields batches of csv files from
+    the subdirectories `class_a` and `class_b`, together with labels
+    0 and 1 (0 corresponding to `class_a` and 1 corresponding to `class_b`).
```

Comment:
So 1 csv file = 1 sample? Is this a common case? Usually you have 1 row = 1 sample.